File: frontend/lib/api/simulation.ts
import { apiPost } from "@/lib/api/http";
import { DEMO_STRATEGIES } from "@/lib/demo";
import type { SimulationConfig, StrategyMetrics, WorldSummary } from "@/types/api";

export type GenerateResponse = {
  run_id: string;
  world_summary: WorldSummary;
  synthetic: boolean;
};

export type RunStrategiesResponse = {
  run_id: string;
  seed: number;
  strategy_results: Record<string, StrategyMetrics>;
  synthetic: boolean;
};

export function generateWorld(config: SimulationConfig) {
  return apiPost<GenerateResponse>("/api/simulator/generate", config);
}

export function runBaselines(runId: string) {
  return apiPost<RunStrategiesResponse>("/api/simulator/run", {
    run_id: runId,
    strategies: [...DEMO_STRATEGIES],
  });
}

File: backend/app/schemas/simulation.py
from pydantic import BaseModel, Field

from app.simulator.config import SimulationConfig


class GenerateResponse(BaseModel):
    run_id: str
    world_summary: dict
    synthetic: bool = True


class RunRequest(BaseModel):
    run_id: str
    strategies: list[str] = Field(
        default_factory=lambda: ["naive", "rule_based", "reclaim"]
    )


class RunResponse(BaseModel):
    run_id: str
    seed: int
    strategy_results: dict
    synthetic: bool = True


class SimulationRunOut(BaseModel):
    run_id: str
    seed: int
    synthetic: bool
    status: str
    config: SimulationConfig
    world_summary: dict
    strategy_results: dict
    created_at: str
    completed_at: str | None
    error: str | None = None
